### PR TITLE
Contact component

### PR DESCRIPTION
**Pull request by ayshesaflo**
issue #25 

Create Contact Us page that contains company contact into and a form
when form's data is submitted, data will be saved in firebase ContactUs collection

web screen
image

tablet screen
image

phone screen
image

### CODE REVIEW

**Review by ceyhungulbas — CHANGES_REQUESTED** [review]
Hey Ayshe! Eline saglik, I just seen some issues:

App.js
- ContactUSPage should be inside of container

ContactUs.jsx
- line 16 there's a console.log
- typo Massege -> Message
- line 50 `<div className="contatctinfo">` typo
- line 76 `<form className="inpurForm" onSubmit={handleSubmit}>` typo

ContactUs.css
- typos in here too

ContactUSPage.jsx
- line 3 and 11 `ConactUSPage()` typo

Bug:
resim
resim

When user add so long text Message area going like endless line. I think we should change to a box.

**Comment by ayshesaflo** [comment]
> * `ConactUSPage()`


The required changes have done

File: src/App.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ import { Container } from "react-bootstrap";
 import "./App.css";
 import Footer from "./components/Footer/Footer";
 import "bootstrap/dist/css/bootstrap.min.css";
+import ContactUSPage from "./Container/ContactUSPage";
 
 function App() {
   return (
     <>
       <Container>
+        <ContactUSPage />
```
Comment (udenizdemirbilek) [inline src/App.js]: Please update App with git pull, then link ContactUSPage as a Route like [here](https://github.com/ReCoded-Org/capstone-turkey-giftify/pull/15/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)

File: src/App.js [hunk 2]
```diff
@@ -3,11 +3,13 @@ import { Container } from "react-bootstrap";
 import "./App.css";
 import Footer from "./components/Footer/Footer";
 import "bootstrap/dist/css/bootstrap.min.css";
+import ContactUSPage from "./Container/ContactUSPage";
```
Comment (udenizdemirbilek) [inline src/App.js]: folder name should be "containers" instead of "Container"

File: src/components/ContactUs/__tests__/ContactUs.test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import ContactUs from "./../ContactUs";
+import { cleanup } from "@testing-library/react";
+import renderer from "react-test-renderer";
+
+afterEach(cleanup);
+
+it("matches Snapshot", () => {
```
Comment (udenizdemirbilek) [inline src/components/ContactUs/__tests__/ContactUs.test.js]: please change test name

**Review by udenizdemirbilek — CHANGES_REQUESTED** [review]
Good work Ayshe!
I've added my comments

**Comment by ayshesaflo** [comment]
@halitbatur PTAL

The required changes are done
the name of Snapshot
App is updated
The "Container" file changed to "containers"

**Review by ceyhungulbas — APPROVED** [review]
I checked it now. LGTM! Ellerine saglik!

- Responsiveness is working
- Functionality is working
- Test message is correct
- Folders name fixed
- App.js Router part fixed

File: src/App.js [hunk 4]
```diff
@@ -14,12 +15,12 @@ function App() {
     <>
       <Container fluid className="mainContainer">
         <Router>
-          <Route path="/" component={Navbar} />
+          <Route ecactpath="/" component={Navbar} />
```
Comment (halitbatur) [inline src/App.js]: please remove the "ecact" word from here

File: src/components/ContactUs/ContactUs.css [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  text-align: center;
+  margin-bottom: 20px;
+}
+
+.contactInfo {
+  color: white;
+  padding: 20px;
+  height: 400px;
+  width: 350px;
+}
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.css:17]: new line needed

File: src/components/ContactUs/ContactUs.css [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+.submitData {
+  align-self: center;
+  width: 100px;
+  background-color: #023e8a;
+  color: white;
+  border-radius: 20px;
+}
+
+.Message {
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.css]: classNames should be camelCased

File: src/components/ContactUs/ContactUs.jsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from "react";
+import { FiMapPin, FiPhone, FiMail } from "react-icons/fi";
+import { ImFacebook2 } from "react-icons/im";
+import { GrInstagram } from "react-icons/gr";
+import db from "../../firebase";
+import "./ContactUs.css";
+
+function ContactUs() {
+  const [inputdata, setInputData] = useState([]);
+  const [inputName, setInputName] = useState("");
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.jsx]: can we move all the inputs to an object ?

File: src/components/ContactUs/ContactUs.jsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+      .then(() => {
+        setInputName("");
+        setInputEmail("");
+        setInputMessage("");
+      });
+  }
+
+  useEffect(() => {
+    const getData = () => {
+      db.collection("contactUs").onSnapshot(function (querySnapshot) {
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.jsx]: why are we fetching the contactUs data here if its not getting used ?

File: src/containers/ContactUSPage/ContactUSPage.jsx [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import ContactUs from "../../components/ContactUs/ContactUs";
```
Comment (halitbatur) [inline src/containers/ContactUSPage/ContactUSPage.jsx]: please add new line after import

File: src/containers/ContactUSPage/ContactUSPage.jsx [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
```
Comment (halitbatur) [inline src/containers/ContactUSPage/ContactUSPage.jsx]: file name is incorrect it should be ContactUsPage

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 11]
```diff
@@ -5,10 +5,12 @@ import { Route } from "react-router";
 import { BrowserRouter as Router } from "react-router-dom";
 import "./App.css";
 import Footer from "./components/Footer/Footer";
+import ContactUSPage from "./containers/ContactUsPage/ContactUsPage";
```
Comment (halitbatur) [inline src/App.js]: the import name is still wrong it needs to be ContactUsPage the s is not capital

File: src/components/ContactUs/ContactUs.css [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+.facebook {
+  color: #0050f1;
+}
+
+@media screen and (min-width: 800px) {
+  .box {
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+  }
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.css:86]: new line between each rule please

File: src/components/ContactUs/ContactUs.css [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+
+  .contactInfo {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-around;
+    text-align: start;
+    width: 250px;
+    height: 300px;
+  }
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.css:118]: new line needed between each rule

File: src/components/ContactUs/ContactUs.css [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.css:147]: new line missing

File: src/components/ContactUs/ContactUs.jsx [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+        }))
+      );
+    });
+  };
+
+  useEffect(() => {
+    getData();
+  }, [inputdata]);
+
+  function handleChange(evt) {
```
Comment (halitbatur) [inline src/components/ContactUs/ContactUs.jsx]: unconsistincy some functions are arrow and others are old please make all of them arrow functions

**Review by halitbatur — CHANGES_REQUESTED** [review]

**Review by halitbatur — APPROVED** [review]
good work!

**Review by udenizdemirbilek — APPROVED** [review]
Good work Ayshe! Well done!

**Review by busesisi — APPROVED** [review]
Ellerine sağlık Ayshe, great work!

**Pull request merged**